services/pushover.py

```diff
@@ -14,19 +22,29 @@
 
 class PushoverError(Exception): pass
 
-def pushover(**kwargs):
-    assert 'message' in kwargs
+def pushover(message, user, token, params, filepayload=None):
+
+    if token is None:
+        params['token'] = os.environ['PUSHOVER_TOKEN']
+    else:
+        params['token'] = token
+    #params['token'] = token
 
-    if not 'token' in kwargs:
-        kwargs['token'] = os.environ['PUSHOVER_TOKEN']
-    if not 'user' in kwargs:
-        kwargs['user'] = os.environ['PUSHOVER_USER']
+    #if user is None:
+    #    params=['user'] = os.environ['PUSHOVER_USER']
+    #else:
+    #    params['user'] = user
+    params["user"] = user
+
+    params["message"] = message
 
     url = urlparse.urljoin(PUSHOVER_API, "messages.json")
-    data = urllib.urlencode(kwargs)
```

Comment:
Don't we still need to URL encode the request parameters?

Reply:
I think that's handled natively by requests in here:
```
if filepayload is None:
  r = requests.post(url,data=params, headers={'User-Agent': 'Python'})
else:
  r = requests.post(url,data=params, files=filepayload, headers={'User-Agent': 'Python'})
```